### ISSUE TITLE
Compilation error on XDP acceleration + ingress controller

### ISSUE THREAD

**Issue by gongysh2004** [cilium/cilium#21381]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I am using the helm values to install my cilium cni:

```root@compute1:~# cat values.yaml
debug:
  enabled: true
externalIPs:
  enabled: true
hostPort:
  enabled: true
hostServices:
  enabled: true
  protocols: tcp
rollOutCiliumPods: true
operator:
  rollOutPods: true
kubeProxyReplacement: strict
k8sServiceHost: 192.168.50.11
k8sServicePort: 6443
nodePort:
  enabled: true
operator:
  replicas: 1
bpf:
  lbExternalClusterIP: false
# autoDirectNodeRoutes: true
ipv4NativeRoutingCIDR: 192.168.50.0/24
tunnel: disabled
loadBalancer:
  algorithm: maglev
  acceleration: native
  mode: dsr
bgp:
  enabled: true
  announce:
    loadbalancerIP: true
ingressController:
  enabled: true
hubble:
  enabled: true
  metrics:
    enabled:
      - dns
      - drop
      - tcp
      - flow
      - port-distribution
      - icmp
      - http
  listenAddress: ":4244"
  relay:
    enabled: true
    rollOutPods: true
  ui:
    enabled: true
    rollOutPods: true

```

but the cilium agent cannot run, in fact, they are in error state.
after that, I digged the logs of one of those agents:

`kubectl  logs -n kube-system cilium-scwm4`

it shows me there are some error there:

```
level=debug msg="Initial list of identities received" subsys=allocator
level=error msg="Failed to compile bpf_xdp.o: exit status 1" compiler-pid=109 linker-pid=110 subsys=datapath-loader
level=debug msg="In file included from /var/lib/cilium/bpf/bpf_xdp.c:38:" subsys=datapath-loader
level=debug msg="/var/lib/cilium/bpf/lib/nodeport.h:1820:11: error: implicit declaration of function 'ctx_redirect_to_proxy_hairpin_ipv4' [-Werror,-Wimplicit-function-declaration]" subsys=datapath-loader
level=debug msg="                        return ctx_redirect_to_proxy_hairpin_ipv4(ctx," subsys=datapath-loader
level=debug msg="                               ^" subsys=datapath-loader
level=debug msg="1 error generated." subsys=datapath-loader
level=fatal msg="Failed to compile XDP program" error="Failed to compile bpf_xdp.o: exit status 1" subsys=datapath-loader
```

### Cilium Version

```
root@compute1:~# cilium version
cilium-cli: v0.12.3 compiled with go1.19 on linux/amd64
cilium image (default): v1.12.1
cilium image (stable): v1.12.2
cilium image (running): v1.12.2
```

### Kernel Version

```
root@compute1:~# uname -a
Linux compute1 5.15.0-48-generic #54~20.04.1-Ubuntu SMP Thu Sep 1 16:17:26 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

### Kubernetes Version

```
root@compute1:~# kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.4+k3s1", GitCommit:"c3f830e9b9ed8a4d9d0e2aa663b4591b923a296e", GitTreeState:"clean", BuildDate:"2022-08-25T03:45:26Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v4.5.4
Server Version: version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.4+k3s1", GitCommit:"c3f830e9b9ed8a4d9d0e2aa663b4591b923a296e", GitTreeState:"clean", BuildDate:"2022-08-25T03:45:26Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/amd64"}
```

### Sysdump

[Uploading cilium-sysdump-20220921-074959.zip…]()

### Anything else?

I am using  vms to install k3s cluster. the nic device driver are virtio
```
root@compute1:~# ethtool -i eth0
driver: virtio_net
version: 1.0.0
firmware-version:
expansion-rom-version:
bus-info: 0000:00:0b.0
supports-statistics: yes
supports-test: no
supports-eeprom-access: no
supports-register-dump: no
supports-priv-flags: no
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pchaigno** [comment]
This is happening because it seems the ingress controller is incompatible with XDP acceleration today. We should add a fatal in that case. cc @borkmann @jrajahalme

**Comment by sayboras** [comment]
I think this should be fixed as part of v1.13.2 branch, as we are having CI coverage for XDP + Ingress controller enabled.

I am closing this issue, please let me know if you think the issue is still happening.

https://github.com/cilium/cilium/blob/29dafd82e73088113469c1d9d6fcce19db493e51/.github/workflows/conformance-ingress.yaml#L43-L46
